src/foam/dao/java/JDAO.js

```diff
@@ -32,11 +37,20 @@ foam.CLASS({
               .setCreateFile(true)
               .build());
 
-            // create a composite journal of repo journal
-            // and runtime journal and load them all
+            /* Create a composite journal of repo journal and runtime journal
+              and load them all.*/
+            X initialStorageX = x;
+            try {
```

Comment:
Do we really need a try/catch for this?  I think a simpler if/else is better than blindly catching all possible exceptions.

Reply:
But it does throw exceptions. Re-throwing doesn't make sense. If the string isn't set or is empty, then it should default to regular rather than fail JDAO completely.

Reply:
The dot.zero file does not need to exist.